backend/test/backend_tests/rpc_team_test.clj

```diff
@@ -467,3 +467,146 @@
+          team        (th/create-team* 1 {:profile-id (:id owner)})
+          proj        (th/create-project* 1 {:profile-id (:id owner)
+                                             :team-id (:id team)})
+          file        (th/create-file* 1 {:profile-id (:id owner)
+                                          :project-id (:id proj)})
+
+          data        {::th/type :create-team-request
+                       ::rpc/profile-id (:id requester)
+                       :file-id (:id file)}
+
+          out   (th/command! data)
+          edata (-> out :error ex-data)]
+
+      ;; request with requester muted should fail
+      (t/is (not (th/success? out)))
+      (t/is (= 0 (:call-count @mock)))
+
+      (t/is (= :validation (:type edata)))
+      (t/is (= :member-is-muted (:code edata)))
+      (t/is (= (:email requester) (:email edata))))))
+
+
+(t/deftest create-team-request-owner-bounce
+  (with-mocks [mock {:target 'app.email/send! :return nil}]
+    (let [owner       (th/create-profile* 1 {:is-active true :email "owner@bar.com"})
+          requester   (th/create-profile* 2 {:is-active true :email "requester@bar.com"})
+          team        (th/create-team* 1 {:profile-id (:id owner)})
+          proj        (th/create-project* 1 {:profile-id (:id owner)
+                                             :team-id (:id team)})
+          file        (th/create-file* 1 {:profile-id (:id owner)
+                                          :project-id (:id proj)})
+
+          pool        (:app.db/pool th/*system*)
+          data        {::th/type :create-team-request
+                       ::rpc/profile-id (:id requester)
+                       :file-id (:id file)}]
+
+
+      (th/create-global-complaint-for pool {:type :bounce :email "owner@bar.com"})
+      (let [out   (th/command! data)
+            edata (-> out :error ex-data)]
+
+        ;; request with owner bounce should fail
+        (t/is (not (th/success? out)))
+        (t/is (= 0 (:call-count @mock)))
+
+        (t/is (= :restriction (:type edata)))
+        (t/is (= :email-has-permanent-bounces (:code edata)))
+        (t/is (= "private" (:email edata)))))))
+
+(t/deftest create-team-request-requester-bounce
+  (with-mocks [mock {:target 'app.email/send! :return nil}]
+    (let [owner       (th/create-profile* 1 {:is-active true :email "owner@bar.com"})
+          requester   (th/create-profile* 2 {:is-active true :email "requester@bar.com"})
+          team        (th/create-team* 1 {:profile-id (:id owner)})
+          proj        (th/create-project* 1 {:profile-id (:id owner)
+                                             :team-id (:id team)})
+          file        (th/create-file* 1 {:profile-id (:id owner)
+                                          :project-id (:id proj)})
+
+          pool        (:app.db/pool th/*system*)
+          data        {::th/type :create-team-request
```

Comment:
create-team-request, looks like a not very clear name, can it be `create-team-access-request` ?